include/utility/dbus_utility.hpp

```diff
@@ -286,5 +286,34 @@ inline types::BiosAttributeCurrentValue
 
     return std::get<1>(l_attributeVal);
 }
+
+/**
+ * @brief API to check if Chassis is powered on.
+ *
+ * This API queries Phosphor Chassis State Manager to know whether
+ * Chassis is powered on.
+ *
+ * @throw std::runtime_error
+ * @return true if chassis is powered on, false otherwise
+ */
+inline bool isChassisPowerOn()
+{
+    auto powerState = dbusUtility::readDbusProperty(
+        "xyz.openbmc_project.State.Chassis",
+        "/xyz/openbmc_project/state/chassis0",
+        "xyz.openbmc_project.State.Chassis", "CurrentPowerState");
+
+    if (auto curPowerState = std::get_if<std::string>(&powerState))
+    {
+        if ("xyz.openbmc_project.State.Chassis.PowerState.On" == *curPowerState)
+        {
+            logging::logMessage("VPD cannot be read in power on state.");
+            return true;
+        }
+        return false;
+    }
+
+    throw std::runtime_error("Dbus call to get chassis power state failed");
```

Comment:
We don;t need to throw from here.
Please refer this.
https://github.com/ibm-openbmc/openpower-vpd-parser/pull/432/files#diff-0cbb4d9bfcabf86d2100f0d6b8bb92a13133ed74f08ac35ee31f27a88fc591aaR115

Reply:
Addressed this.